wal.go

```diff
@@ -79,17 +79,20 @@ type Options struct {
 	// Perms represents the datafiles modes and permission bits
 	DirPerms  os.FileMode
 	FilePerms os.FileMode
+	// RecoverCorruptedTail will attempt to recover a corrupted tail in the last segment automatically.
+	RecoverCorruptedTail bool
```

Comment:
iiuc, the following scenarios would also need to be handled for completeness: 

- partial writes of payload, following the offset (i.e. when `s.ebuf = ebuf[:pos]` assignment below panics)
- finding the last segment with non-corrupted data (the existing last segment file may only have corrupted data if e.g. it's 0 byte or the first entry is corrupted). We should delete that file and attempt reading second-to-last one (preferably with recoverCorruptedTail=false)
- the last segment post-recovery must have non-zero entries or whatever is appropriate considering https://github.com/tidwall/wal/pull/31

Reply:
thanks for the feedback, will think it through tmr, it's late here ;D